### ISSUE TITLE
Configure tooltip max width

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/vscode#14165]
When you hover over a type in a TypeScript file, the tooltip that appears has a very low max width, making it difficult to read long type signatures.

**Comment by OliverJAsh** [comment]
I would include a screenshot but it's not easy to take a screenshot keeping the tooltip on screen!

**Comment by maxhoffmann** [comment]
![2017-01-03 14 38 46](https://cloud.githubusercontent.com/assets/1227441/21610898/75c0566a-d1c2-11e6-9898-4239d5a34c63.gif)

Errors are quite unreadable if the width does not adapt to the font size.

**Comment by rakkarage** [comment]
![2017-02-16 2](https://cloud.githubusercontent.com/assets/962340/23047121/d3344064-f47b-11e6-9f0f-7ec6841cb7ff.png)
it would be nice if the tip used the available space instead of wrapping at an arbitrary point

**Comment by rakkarage** [comment]
![2017-03-09 2](https://cloud.githubusercontent.com/assets/962340/23755841/b0bc60f4-04af-11e7-8f48-df0da13ebfcb.png)

**Comment by anowell** [comment]
![tooltip](https://cloud.githubusercontent.com/assets/2629687/25759705/d0fe8384-3190-11e7-8061-8b2e4e61ee63.png)

Same problem using rust's language server.

**Comment by crucialfelix** [comment]
General styling control would be very helpful. Also line-height.  

Another approach would be to have a bottom panel that displays the full error and the tooltip displays a shortened version.

**Comment by rakkarage** [comment]
![2018-02-15](https://user-images.githubusercontent.com/962340/36287949-29004eba-1286-11e8-89da-22a114bb2f2c.png)

**Comment by Tiedye** [comment]
![image](https://user-images.githubusercontent.com/6187173/37847674-0187f43a-2ea8-11e8-9211-8105f92d1622.png)

When using the auto import feature it is often impossible to differentiate whats what as the import path is cut off due to the limited width of the suggestions window.

**Comment by MikhailArkhipov** [comment]
Same in Python. We have to just through hoops trying to flow doc nicely. And when it appears OK on Windows, it breaks on Mac... 

![image](https://user-images.githubusercontent.com/12820357/38449834-94ddf334-39c9-11e8-8646-6eb6729a9951.png)

And it is inconsistent. Ex, hover is OK

![image](https://user-images.githubusercontent.com/12820357/38449878-db4a38fa-39c9-11e8-9c95-ed5b51a5fdb6.png)

But completion list decides to wrap. What gives?

![image](https://user-images.githubusercontent.com/12820357/38449890-0bc533f4-39ca-11e8-951f-8eb4bc0d7aad.png)

**Comment by rakkarage** [comment]
![2018-04-15_li](https://user-images.githubusercontent.com/962340/38780926-a8ab6656-40ab-11e8-93c7-21170b8f5529.jpg)

**Comment by KaelWD** [comment]
Someone say long type signatures?
![image](https://user-images.githubusercontent.com/16421948/39526258-94e2689a-4e61-11e8-818c-a06e3a05138d.png)

**Comment by ShenHongFei** [comment]
Here is a workaround by hacking the file "C:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\workbench.main.css"

```css
    /* suggest-widget size */
        .monaco-editor .suggest-widget.docs-side {
            width: 1600px;
        }
        .monaco-editor .suggest-widget.docs-side > .details {
            width: 70%;
            max-height: 800px !important;
        }
        .monaco-editor .suggest-widget.docs-side > .tree {
            width: 30%;
            float: left;
        }
        
    /* parameter-hints-widget */
        .editor-widget.parameter-hints-widget.visible {
            max-height: 800px !important;
        }
        .monaco-editor .parameter-hints-widget > .wrapper {
            max-width: 1600px;
        }
```

Simply append the code above after the original content of file, and restart VSCode. (Title bar may alert a warning message "Not Supported" and hacked changes would be overridden by updating VSCode)

![0062](https://user-images.githubusercontent.com/15074472/42794438-59789898-89b1-11e8-9480-84c8e9816ba7.png)

**Comment by 7sharp9** [comment]
It would be nice to have some control on the width.  

The above does work but vscode presents the message your installation may be corrupt on restarting.

**Comment by joealden** [comment]
What is preventing the devs from providing a configuration option for this? This issue has been open for almost 2 years, and it bugs me on a daily basis. Would love to see it fixed.

**Comment by tecosaur** [comment]
@roblourens (since you seem like the most relevant person to ping) is anything likely to be done about this?

**Comment by Astrantia** [comment]
@roblourens I would love to hear an explanation from devs what are the difficulties of implementing this. I imagine it's settings a dynamic max-width and adding some more lines to add it in Settings. Are there other design blockers?

**Comment by Astrantia** [comment]
@rebornix Will you take a look at this?

**Comment by rebornix** [comment]
Assigned to @joaomoreno and @misolori as I remember we are going to have some UX improvements for editor tooltips and so on.

**Comment by Astrantia** [comment]
@joaomoreno @misolori @rebornix maybe now is the time to take this on? current iteration tries to solve parameter hints ux and this is not that much different

**Comment by drewlustro** [comment]
100% agree. This is just a workaround.

It's hard enough to read TypeScript lint as it is! :)

**Comment by lautaroem1** [comment]
Currenlty using VSCode 1.30.2, followed @drewlustro instructions step by step, but it's not working.
I don't even get the _VSCode corrupt_ error. I'm running Windows 10 (V.1803).

**Comment by tamuratak** [comment]
This issue seems to be fixed in 1.32.0-insider (1.32.0-insider) 2e4d4a6bb16a65303b460795fa4ac734f9aea43e. I hope this is an intended behavior. 

I want to thank the developers.

**Comment by tamuratak** [comment]
This change is related to #67076.

I confirmed that the max width of `CompletionItem.documentation` is still small as mentioned above.

<img width="867" alt="2019-02-22 17 06 19" src="https://user-images.githubusercontent.com/10665499/53228400-7111e980-36c4-11e9-91f6-b77603783fba.png">

**Comment by tamuratak** [comment]
Update: in #69388,  only the hovers for problems and the hovers with code blocks become to have larger width. Others have smaller width.

**Comment by JoeyRxy** [comment]
@ShenHongFei Hello there,your solution is great,but I want to ask how to get rid of the word `[Unsupported]` at the title.As sort of OCD,sufferring from that.

thx a lot!

**Comment by szhu** [comment]
> 沈鸿飞 Hello there,your solution is great,but I want to ask how to get rid of the word `[Unsupported]` at the title.As sort of OCD,sufferring from that.

This extension was created specifically for solving this: https://marketplace.visualstudio.com/items?itemName=lehni.vscode-fix-checksums

**Comment by SheepDomination** [comment]
> This issue seems to be fixed in 1.32.0-insider (1.32.0-insider) [2e4d4a6](https://github.com/microsoft/vscode/commit/2e4d4a6bb16a65303b460795fa4ac734f9aea43e). I hope this is an intended behavior.
> 
> I want to thank the developers.

The `VS Code is corrupt error` is fixed ?

**Comment by jpike88** [comment]
This has been a long standing annoyance, with an issue that's like 2 and a half year ago. It shouldn't take much effort to have a config property that overrides the width of the tooltip, surely? Like, 10 mins of effort?

<img width="525" alt="Screen Shot 2019-07-18 at 6 41 15 PM" src="https://user-images.githubusercontent.com/9585787/61454671-b1337a80-a98b-11e9-8f7e-91d95e475094.png">

**Comment by szhu** [comment]
@jpike88 if it's really 10 mins of effort, then I'm sure you can submit a PR for this? I definitely can't do this in anywhere close to 10 mins, so it would be much appreciated!

**Comment by GitClickOk** [comment]
Seriously, I want instead of this:
![Capture 1735](https://user-images.githubusercontent.com/19438005/83681730-2cf4e080-a5b9-11ea-8752-a150bbad42b3.png)

At least something like this (with syntax highlighting it will be even better):
![Capture 1736](https://user-images.githubusercontent.com/19438005/83681769-3ed68380-a5b9-11ea-9957-d9c661e1d1ad.png)

**Comment by burner1024** [comment]
> @jpike88 if it's really 10 mins of effort, then I'm sure you can submit a PR for this? I definitely can't do this in anywhere close to 10 mins, so it would be much appreciated!

There's already been a PR.

I really don't understand this. I figure most VScode devs develop VScode in VScode, are they comfortable with the current state?

**Comment by misolori** [comment]
@jrieken fyi, it would be great if the hover could be resized [like the suggest widget](https://github.com/microsoft/vscode/issues/29126)

**Comment by modo-lv** [comment]
How is this issue 4 years old and such a basic thing still isn't implemented?

**Comment by dons20** [comment]
Is there any word on whether we could see this added soon? The suggestion widget has had resizing enabled since v1.51 in October, 2020. Is this tooltip widget considerably harder to add this feature for or are there additional considerations to be made? Whether it's a click to resize, or a manual adjustment in settings, could we have an update on the status of this addition? 

Opening new editor tabs just to properly read code annotations for some more descriptive components on my large monitors does feel like an unnecessary extra step.

**Comment by arash-bizcover** [comment]
https://github.com/microsoft/vscode/issues/69388

**Comment by arash-bizcover** [comment]
https://github.com/microsoft/vscode/issues/69835

**Comment by arash-bizcover** [comment]
@jrieken might be very related and close to what you did here:
https://github.com/microsoft/vscode/pull/109094

**Comment by GilShoshan94** [comment]
Hi,
Instead of configuring the max width as the title says, could the tooltip hover be resizable [like the suggest widget](https://code.visualstudio.com/updates/v1_51#_resizable-suggestions)  ?

(Also check #29126 for more info on what was done with the suggest widget)

**Comment by Arxareon** [comment]
It's sad to see that it's still not a thing.. I would really like to adjust the height of hover description tooltips.. The big brother Visual Studio does a much better job already. Let VS Code catch up a little..

**Comment by suspectinside** [comment]
another possible view for this issue https://github.com/microsoft/vscode/issues/153984

**Comment by thisentirecitymustbepurged** [comment]
It is 2022/11/11. Christmas is close and I made sure Santa knows what I want!

**Comment by TFLXX** [comment]
"Excuse me, it's 2023" 🙂

**Comment by a-saven** [comment]
This one will be my favorite. Copilot suggestions and TS errors just can't be seen in this tooltip. It's time to fix it.

**Comment by ryneandal** [comment]
If this is still open and no one is looking at it, I'll take a crack at it. Feel free to assign it to me if that's alright.

**Comment by aiday-mar** [comment]
Hey all, thank you for your patience. A PR has been made which is currently under work but nearing its completion. You may find it here: https://github.com/microsoft/vscode/pull/176744. Once this PR is merged, you will be able to resize content hovers within Insiders.

**Comment by drewlustro** [comment]
Seven years ago, we all had a dream.
Then Aiday descended.
She designed the world. She became the architect.
Now we don't worry about waking up.
Because Aiday got that dawg in her.

**Comment by svipas** [comment]
@aiday-mar Would you add "Reset Hover Widget Size" command in Command Palette? Like we have for suggest widget: "Reset Suggest Widget Size". It would be good if we could reset the size to the initial one.

**Comment by MarvellousLunatic** [comment]
Will this feature also allow us to define the window size of a hover provider in a extension above the current limitations?

**Comment by aiday-mar** [comment]
Hi @svipas, if is is decided that the hover widget size should be configurable, there will be a command to clear the persisted sizes. This feature is still under discussion.

Hi @ShalokShalom, depending on the implementation, it is possible that the persisted sizes are persisted or not across different sessions. This is still under discussion. Currently the implementation is such that the maximum _natural_ width is the same as before. 

Hi @MarvellousLunatic, the current implementation does not provide this functionality. This could be addressed in a separate issue as it is related to the usage of the hover widget by extensions. 

The current PR that is worked on is the following: https://github.com/microsoft/vscode/pull/178811

**Comment by ShalokShalom** [comment]
Thanks a lot. And do you consider increasing the default size, that is unfitting to most real world scenarios? 

The windows is so small, that I feel the need to increase it even on my WXGA Laptop.

**Comment by aiday-mar** [comment]
Hey all, thank you for the patience. A PR has been merged in order to make the content hover resizable. The change should be visible in tomorrow's VS Code insiders version. There currently is no mechanism to save the content hover - we are discussing how best to approach it. You may track the issue on persisting the hover content here: https://github.com/microsoft/vscode/issues/185743

**Comment by macintacos** [comment]
Just want to say thank you @aiday-mar, you seem to have been tackling a lot of these high-level nitpicks that have been in the backlog for years, and for that I am very appreciative.

**Comment by svipas** [comment]
@aiday-mar I can't find a command to clear/reset hover widget size, is this has been added?

**Comment by aiday-mar** [comment]
Hi @svipas, currently the command does not exist because a persisting mechanism has not been put into place as of yet. It will appear when the latter is added.
